### PR TITLE
Add Clear All action with confirmation to canvas

### LINKED ISSUE #125: Clear Canvas with Confirmation Dialog
**Difficulty:** Medium
**Area:** frontend

## Description

TensorMap's canvas currently has no way to quickly clear all nodes and edges and start fresh. When users want to build a new model from scratch, they must manually select and delete each node individually. This is tedious for large graphs.

This issue adds a "Clear All" button to the canvas toolbar area that removes all nodes and edges from the ReactFlow canvas. Since this is a destructive action, a confirmation dialog should appear before clearing to prevent accidental data loss.

## Acceptance Criteria

- [ ] A "Clear All" button is visible on the canvas page
- [ ] The button is disabled when the canvas has no nodes
- [ ] Clicking the button opens a confirmation dialog
- [ ] The dialog shows how many nodes will be removed
- [ ] Clicking "Cancel" in the dialog closes it without clearing
- [ ] Clicking "Clear All" in the dialog removes all nodes and edges from the canvas
- [ ] After clearing, the properties panel shows its default empty state
- [ ] The button and dialog use consistent styling with the rest of the application

## Resources

- `tensormap-frontend/src/components/DragAndDropCanvas/Canvas.jsx` — ReactFlow canvas with `useNodesState` and `useEdgesState`
- `tensormap-frontend/src/components/DragAndDropCanvas/NodePropertiesPanel.jsx` — properties panel (should handle empty selection)
- `tensormap-frontend/src/components/ui/dialog.jsx` — shadcn/ui Dialog component
- `tensormap-frontend/src/components/ui/button.jsx` — shadcn/ui Button component
- [ReactFlow useNodesState](https://reactflow.dev/api-reference/hooks/use-nodes-state)
- [Lucide React icons](https://lucide.dev/icons/)

### PR DESCRIPTION
**Pull request by Ananya44444**
## Description

Brief summary of the changes. Reference any related issues.

Fixes #125 

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation update

## How Has This Been Tested?

Describe the tests you ran to verify your changes.

- [x] Existing tests pass
- [ ] New tests added
- [x] Manual testing

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review
- [x] I have added/updated documentation as needed
- [x] My changes generate no new warnings
- [x] Tests pass locally

Added a “Clear All” toolbar action above the ReactFlow canvas using the existing shadcn Button and Dialog components plus a Lucide trash icon.
Show a confirmation dialog before clearing, including the current node count and a warning that the action cannot be undone.
Clearing removes all nodes and edges and resets selection, so the NodePropertiesPanel falls back to its default empty “Save Model” state.

### CODE REVIEW

File: tensormap-frontend/src/components/DragAndDropCanvas/Canvas.jsx [hunk 1]
```diff
@@ -324,6 +334,15 @@ function Canvas() {
     [reactFlowInstance, setNodes],
   );
 
+  const nodeCount = nodes.length;
+  const handleClearAll = useCallback(() => {
+    setNodes([]);
```
Comment (rajutkarsh07) [inline tensormap-frontend/src/components/DragAndDropCanvas/Canvas.jsx]: add 
```
setModelName("");
```
here

**Comment by rajutkarsh07** [comment]
just fix one small change and fix conflicts

File: tensormap-frontend/src/components/DragAndDropCanvas/Canvas.jsx [hunk 2]
```diff
@@ -392,51 +393,46 @@ function Canvas() {
       <div className="flex gap-4">
         <ReactFlowProvider>
           <Sidebar />
```
Comment (ivantha) [inline tensormap-frontend/src/components/DragAndDropCanvas/Canvas.jsx]: **[MEDIUM]** Visual regression — border and rounded corners silently removed from canvas wrapper

The original canvas wrapper included `rounded-md border`, giving the canvas a visible bounding box that matched the project's design system. These classes were silently dropped in the refactor:

```
// Before
<div className="min-w-0 h-[62vh] flex-1 rounded-md border" ...>

// After
<div className="h-[62vh] flex-1" ...>
```

This is a purely visual regression that is not mentioned in the PR description and was not an explicit goal of the "Clear All" feature. It changes the look of every canvas view in the application.

**Suggestion:**
```
If removing the border is intentional, call it out explicitly in the PR description and get design sign-off. Otherwise, restore the classes:
```jsx
<div className="min-w-0 h-[62vh] flex-1 rounded-md border" ref={reactFlowWrapper}>
```
```

File: tensormap-frontend/src/components/DragAndDropCanvas/Canvas.jsx [hunk 3]
```diff
@@ -361,6 +353,16 @@ function Canvas() {
     [reactFlowInstance, setNodes],
   );
 
+  const nodeCount = nodes.length;
+  const handleClearAll = useCallback(() => {
+    setNodes([]);
+    setEdges([]);
+    setModelName("");
+    setSelectedNodeId(null);
+    closeContextMenu();
```
Comment (ivantha) [inline tensormap-frontend/src/components/DragAndDropCanvas/Canvas.jsx]: **[MEDIUM]** No tests added for the Clear All confirmation flow

The PR checklist acknowledges that "New tests added" is unchecked. The Clear All flow has several distinct code paths that should be covered by automated tests:

1. Dialog does **not** open when `nodeCount === 0` (button is disabled).
2. Clicking the button opens the dialog and displays the correct node count.
3. Clicking **Cancel** closes the dialog without modifying `nodes`, `edges`, `modelName`, or `selectedNodeId`.
4. Clicking **Clear** calls `setNodes([])`, `setEdges([])`, resets both string states, calls `closeContextMenu`, and closes the dialog.
5. Pluralisation edge cases: exactly 1 node vs. 0/2+ nodes.

Without these tests, regressions in the confirmation flow will go undetected.

**Suggestion:**
```
Add a test file (e.g., `Canvas.test.jsx`) using React Testing Library:
```jsx
it('does not open dialog when canvas is empty', () => {
  render(<Canvas />);
  expect(screen.getByRole('button', { name: /clear all/i })).toBeDisabled();
});

it('opens dialog with correct node count on click', async () => {
  // seed nodes ...
  userEvent.click(screen.getByRole('button', { name: /clear all/i }));
  expect(await screen.findByText(/3 nodes/i)).toBeInTheDocument();
});

it('clears canvas on confirm', async () => { /* ... */ });
it('keeps canvas on cancel', async () => { /* ... */ });
```
```

File: tensormap-frontend/src/components/DragAndDropCanvas/Canvas.jsx [hunk 4]
```diff
@@ -392,51 +393,46 @@ function Canvas() {
...
+              onEdgesChange={onEdgesChange}
+              onConnect={onConnect}
+              onInit={setReactFlowInstance}
+              onDrop={onDrop}
+              onDragOver={onDragOver}
+              onNodeClick={onNodeClick}
+              onPaneClick={onPaneClick}
+              onNodeContextMenu={onNodeContextMenu}
+              nodeTypes={nodeTypes}
+              defaultViewport={defaultViewport}
```
Comment (ivantha) [inline tensormap-frontend/src/components/DragAndDropCanvas/Canvas.jsx]: **[LOW]** Clear All trigger button changed from `destructive` to `outline` — weakens affordance

The trigger button that opens the destructive-action confirmation dialog was changed from `variant="destructive"` to `variant="outline"`. The `size="sm"` prop was also dropped.

Using a neutral `outline` button for an action whose sole purpose is to destroy all canvas content is a weaker affordance. Standard UX guidance (Material Design, Apple HIG) recommends that the entry point to a destructive flow carries visual weight (red / destructive styling) so the user is primed for what follows before they even see the confirmation dialog. The confirmation dialog's own "Clear" button correctly remains `variant="destructive"`.

**Suggestion:**
```
Restore destructive styling on the trigger:
```jsx
<Button
  variant="destructive"
  size="sm"
  onClick={() => setClearAllOpen(true)}
  disabled={nodeCount === 0}
>
  <Trash2 className="mr-2 h-4 w-4" />
  Clear All
</Button>
```
```

**Review by ivantha — COMMENTED** [review]
## ◎ Argus Review

The PR successfully delivers the Clear All confirmation dialog and consolidates the toolbar buttons into a ReactFlow Panel, with clean state variable naming and readable pluralisation. However, the layout refactor introduces two unintentional regressions: the removal of `min-w-0` (risking flex overflow in the canvas row) and the loss of the `rounded-md border` visual styling — neither of which is mentioned in the PR description. The confirmation dialog also drops the "This action cannot be undone" warning, weakening the UX of an irreversible destructive action. The `useCallback` dependency array is internally inconsistent, and no automated tests cover the new confirmation flow.

**Comment by ivantha** [comment]
This branch is behind `main`. Please rebase onto the latest `main` and push an update so CI can re-run and the merge can proceed.

**Comment by ivantha** [comment]
Thank you for the Clear All feature! This functionality was already shipped in merged PR #169 which added a Clear All button with confirmation dialog to the canvas. This PR is now superseded — feel free to contribute other improvements!
